### PR TITLE
Generate type for NAME, STRING and NUMBER variables

### PR DESCRIPTION
**Pull request by lysnikolaou**
We can now avoid unnecessary casts in the grammar.

### CODE REVIEW

File: pegen/c_generator.py [hunk 1]
```diff
@@ -492,6 +492,8 @@ def add_var(self, node: NamedItem, names: List[str]) -> Tuple[str, Optional[str]
                     type = rule.type
             elif name.startswith("_loop") or name.startswith("_gather"):
                 type = "asdl_seq *"
```
Comment (gvanrossum) [inline pegen/c_generator.py]: Please use exact comparisons (e.g. ‘name_var’).

**Comment by lysnikolaou** [comment]
Done!

File: pegen/c_generator.py [hunk 2]
```diff
@@ -492,7 +492,7 @@ def add_var(self, node: NamedItem, names: List[str]) -> Tuple[str, Optional[str]
...
-            elif name.startswith("name") or name.startswith("string") or name.startswith("number"):
+            elif name == "name_var" or name == "string_var" or name == "number_var":
```
Comment (gvanrossum) [inline pegen/c_generator.py]: This is better written as `name in ("name_var", ...)`.

Comment (lysnikolaou) [reply pegen/c_generator.py]: Right!

**Pull request merged**